Core/Core/Common/CommonModels/Analytics/AnalyticsMetadataInteractor.swift

```diff
@@ -24,22 +24,33 @@ public protocol AnalyticsMetadataInteractor {
 }
 
 public class AnalyticsMetadataInteractorLive: AnalyticsMetadataInteractor {
+
+    private struct UserMetadata {
+        let uuid: String?
+        var locale: String?
+        let accountUUID: String?
+    }
+
     public init() {}
 
     public func getMetadata() async throws -> AnalyticsMetadata {
-        let flagsStore = ReactiveStore(useCase: GetEnvironmentFeatureFlags(context: Context.currentUser))
+        let flagEnabledStore = ReactiveStore(useCase: GetEnvironmentFeatureFlags(context: Context.currentUser))
             .getEntities()
-            .compactMap { $0 }
+            .compactMap { $0.isFeatureEnabled(.account_survey_notifications) }
 
         let userStore = ReactiveStore(useCase: GetSelfUserIncludingUUID())
             .getEntities(ignoreCache: true)
-            .map { $0.first }
-            .compactMap { $0 }
+            .compactMap { $0.first }
+            .map {
+                UserMetadata(uuid: $0.uuid, locale: $0.locale, accountUUID: $0.accountUUID)
+            }
```

Comment:
It would be better to use `tryMap` instead of `compactMap` and `map` because if `$0.first` is nil then the stream will publish no values (we can just throw in internalerror when profile is nil). The logic in`asyncPublisher` only resumes the continuation if the stream publishes a value so in our case it will wait forever essentially blocking a thread.